ProcessLib/HeatTransportBHE/BHE/BHE_CXA.h

```diff
@@ -0,0 +1,151 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include <Eigen/Eigen>
+
+#include "BaseLib/Error.h"
+
+#include "BHECommon.h"
+#include "FlowAndTemperatureControl.h"
+#include "PipeConfigurationCXA.h"
+
+namespace ProcessLib
+{
+namespace HeatTransportBHE
+{
+namespace BHE
+{
+/**
+ * The BHE_CXA class is the realization of Coaxial pipe with Annular type of the
+ * Borehole Heate Exchanger. In this class, the pipe heat capacity,
+ * pipe heat conductiion, pie advection vectors are intialized according to the
+ * geometry of CXA type of BHE. For CXA type of BHE, 3 primary unknowns are
+ * assigned on the 1D BHE elements. They are the temperature in inflow pipe
+ * T_in, temperature in outflow pipe T_out, temperature of the grout zones
+ * sorrounding the inflow pipe T_g. These primary varaibles are solved
+ * according to heat convection and conduction equations on the pipes and also
+ * in the grout zones. The interaction of the CXA type of BHE and the
+ * sorrounding soil is regulated through the thermal resistance values, which
+ * are calculated specifically during the initialization of the class.
+ */
+class BHE_CXA final : public BHECommon
+{
+public:
+    BHE_CXA(BoreholeGeometry const& borehole,
+            RefrigerantProperties const& refrigerant,
+            GroutParameters const& grout,
+            FlowAndTemperatureControl const& flowAndTemperatureControl,
+            PipeConfigurationCXA const& pipes);
+
+    static constexpr int number_of_unknowns = 3;
+    static constexpr int number_of_grout_zones = 1;
+
+    std::array<double, number_of_unknowns> pipeHeatCapacities() const;
+
+    std::array<double, number_of_unknowns> pipeHeatConductions() const;
+
+    std::array<Eigen::Vector3d, number_of_unknowns> pipeAdvectionVectors()
+        const;
+
+    template <int NPoints, typename SingleUnknownMatrixType,
+              typename RMatrixType, typename RPiSMatrixType,
+              typename RSMatrixType>
+    void assembleRMatrices(
```

Comment:
For discussion: Can this be moved to the local assembler? Maybe after the merge of this PR.